tests/e2e/testsuites/testsuites.go

```diff
@@ -0,0 +1,178 @@
+	t.persistentVolumeClaim, err = client.CoreV1().PersistentVolumeClaims(namespace.Name).Get(t.persistentVolumeClaim.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	// Get the bound PersistentVolume
+	By("validating provisioned persistent volume")
+	t.persistentVolume, err = client.CoreV1().PersistentVolumes().Get(t.persistentVolumeClaim.Spec.VolumeName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	// Check sizes
+	expectedCapacity := t.requestedPersistentVolumeClaim.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
+	claimCapacity := t.persistentVolumeClaim.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
+	Expect(claimCapacity.Value()).To(Equal(expectedCapacity.Value()), "claimCapacity is not equal to requestedCapacity")
+
+	pvCapacity := t.persistentVolume.Spec.Capacity[v1.ResourceName(v1.ResourceStorage)]
+	Expect(pvCapacity.Value()).To(Equal(expectedCapacity.Value()), "pvCapacity is not equal to requestedCapacity")
+
+	// Check PV properties
+	By("checking the PersistentVolume")
+	expectedAccessModes := t.requestedPersistentVolumeClaim.Spec.AccessModes
+	Expect(t.persistentVolume.Spec.AccessModes).To(Equal(expectedAccessModes))
+	Expect(t.persistentVolume.Spec.ClaimRef.Name).To(Equal(t.persistentVolumeClaim.ObjectMeta.Name))
+	Expect(t.persistentVolume.Spec.ClaimRef.Namespace).To(Equal(t.persistentVolumeClaim.ObjectMeta.Namespace))
+	Expect(t.persistentVolume.Spec.PersistentVolumeReclaimPolicy).To(Equal(*t.storageClass.ReclaimPolicy))
+	Expect(t.persistentVolume.Spec.MountOptions).To(Equal(t.storageClass.MountOptions))
+}
+
+func (t *StorageClassTest) Cleanup(client clientset.Interface, namespace *v1.Namespace) {
+	framework.Logf("deleting PersistentVolumeClaim %q/%q", namespace.Name, t.persistentVolumeClaim.Name)
+	err := client.CoreV1().PersistentVolumeClaims(namespace.Name).Delete(t.persistentVolumeClaim.Name, nil)
+	framework.ExpectNoError(err)
+	// Wait for the PV to get deleted if reclaim policy is Delete. (If it's
+	// Retain, there's no use waiting because the PV won't be auto-deleted and
+	// it's expected for the caller to do it.) Technically, the first few delete
+	// attempts may fail, as the volume is still attached to a node because
+	// kubelet is slowly cleaning up the previous pod, however it should succeed
+	// in a couple of minutes.
+	if t.persistentVolume.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
+		By(fmt.Sprintf("deleting the claim's PV %q", t.persistentVolume.Name))
+		err = framework.WaitForPersistentVolumeDeleted(client, t.persistentVolume.Name, 5*time.Second, 10*time.Minute)
+		framework.ExpectNoError(err)
+	}
+	framework.Logf("deleting storage class %s", t.storageClass.Name)
+	err = client.StorageV1().StorageClasses().Delete(t.storageClass.Name, nil)
+	framework.ExpectNoError(err)
+}
+
+func NewStorageClassTestPod(command string) *StorageClassTestPod {
+	return &StorageClassTestPod{
+		pod: &v1.Pod{
+			TypeMeta: metav1.TypeMeta{
```

Comment:
Again, I am 99% sure this is not needed.

Reply:
Copy pasted from upstream code. :)